north/src/runtime/error.rs

```diff
@@ -0,0 +1,130 @@
+            InstallFailure::SignatureNotFound => InstallationResult::SignatureNotFound,
+            InstallFailure::InvalidSignatureYaml => InstallationResult::InvalidSignatureYaml,
+            InstallFailure::MalformedSignature => InstallationResult::MalformedSignature,
+            InstallFailure::MalformedHashes => InstallationResult::MalformedHashes,
+            InstallFailure::MalformedManifest(s) => InstallationResult::MalformedManifest(s),
+            InstallFailure::VerityProblem(s) => InstallationResult::VerityProblem(s),
+            InstallFailure::ArchiveError(s) => InstallationResult::ArchiveError(s),
+            InstallFailure::DeviceMapperProblem(e) => {
+                InstallationResult::DeviceMapperProblem(format!("{}", e))
+            }
+            InstallFailure::LoopDeviceError(e) => {
+                InstallationResult::LoopDeviceError(format!("{}", e))
+            }
+            InstallFailure::HashInvalid(s) => InstallationResult::HashInvalid(s),
+            InstallFailure::KeyNotFound(s) => InstallationResult::KeyNotFound(s),
+            InstallFailure::ApplicationAlreadyInstalled(_) => {
+                InstallationResult::ApplicationAlreadyInstalled
+            }
+            InstallFailure::InternalError(s) => InstallationResult::InternalError(s),
+            InstallFailure::MountError(s) => InstallationResult::MountError(s),
+            InstallFailure::NoVerityHeader => InstallationResult::NoVerityHeader,
+            InstallFailure::UnexpectedVerityAlgorithm(s) => {
+                InstallationResult::UnexpectedVerityAlgorithm(s)
+            }
+            InstallFailure::UnexpectedVerityVersion(n) => {
+                InstallationResult::UnexpectedVerityVersion(n)
+            }
+        }
+    }
+}
+
+#[derive(Error, Debug)]
+pub enum DeviceMapperError {
+    #[error("Failure opening file for device mapper")]
+    OpenDmError,
+    #[error("Failure issuing an IO-CTL call")]
+    IoCtrlError,
+    #[error("Response DM buffer requires too much space")]
+    BufferFull,
+    #[error("Could not create device")]
+    CreateDeviceFailed,
+    #[error("Failure to remove device")]
+    DeviceRemovalFailed,
+    #[error("Failure to suspend device")]
+    SuspendDeviceError,
+}
+
+#[derive(Error, Debug)]
+pub enum LoopDeviceError {
+    #[error("Control file for loop device could not be created")]
+    ControlFileError,
+    #[error("Failure to find or allocate free loop device")]
+    NoFreeDeviceFound,
+    #[error("Failure adding new loop device")]
+    DeviceAlreadyAllocated,
+    #[error("Failure to associate loop device with open file")]
+    AssociateError,
+    #[error("Set Loop status exceeded number of retries ({0})")]
+    StatusWriteBusy(usize),
+    #[error("Set Loop status failed")]
+    SetStatusError,
+    #[error("Failure to set DIRECT I/O mode")]
+    DirectIoError,
+    #[error("Failure to dis-associate loop device from file descriptor")]
+    ClearError,
+}
```

Comment:
The enum entries in error normally do not end with a `Error` because the work Error is already in the enum name.
e.g

`LoopDeviceError::ControlFileError` -> `LoopDeviceError::ControlFile`